amethyst_core/src/hide_hierarchy_system.rs

```diff
@@ -0,0 +1,328 @@
+use crate::{
+    ecs::*
+};
+
+use crate::HiddenPropagate;
+use crate::transform::{Children, Parent};
+use std::collections::HashSet;
+
+/// This system adds a [HiddenPropagate](struct.HiddenPropagate.html)-component to all children
+/// of an entity with a [HiddenPropagate](struct.HiddenPropagate.html) and removes it when it is removed
+/// from the parent.
+pub fn build() -> impl Runnable {
```

Comment:
This is awesome! Thanks for adding tests 👍🏻   Way better than my attempt in #2536